src/lib/core/utils/study-metadata.ts

```diff
@@ -31,3 +32,23 @@ export function makeEntityDisplayName(entity: StudyEntity, isPlural: boolean) {
     ? entity.displayName
     : entity.displayNamePlural ?? `${entity.displayName}s`;
 }
+
+// Traverse down the entities and return an array of collection variables.
+export function findCollections(entity: StudyEntity) {
+  // Create an array of collections, where each collection element is a CollectionVariableTreeNode
+  // that includes that collection's entity id and display name
+  const collections = Array.from(
+    preorder(entity, (e) => e.children ?? [])
+  ).flatMap((e) => {
+    const collectionWithEntity = e.collections?.map((collection) => {
+      return {
+        ...collection,
+        entityId: e.id,
+        entityDisplayName: e.displayName,
+      };
+    });
+    return collectionWithEntity;
```

Comment:
You should avoid returning `undefined` here. Otherwise, you will end up with an array that has `undefined` elements. Returning an empty array ensures that the resulting array (`collections`) does not have `undefined` elements. This will allow you to remove some defensive logic where the return value of this function is used.
```suggestion
    return collectionWithEntity ?? [];
```